generators/python/src/fern_python/generators/context/pydantic_generator_context_impl.py

```diff
@@ -53,6 +53,17 @@ def __init__(
             ):
                 self._non_union_self_referencing_type_ids.add(id)
 
+        self._non_union_types_with_self_referencing_dependencies: Dict[str, Set[str]] = dict()
```

Comment:
why do you need to make this a dict?

Reply:
Discussed offline, but it's just making it a smaller lookup, vs taking the field then getting the dependent types then doing the same lookup